src/node/txreconciliation.cpp

```diff
@@ -237,33 +274,57 @@ class TxReconciliationTracker::Impl
         // However, exploiting (2) should not prevent us from relaying certain transactions.
         //
         // Transactions which don't make it to the set due to the limit are announced via fan-out.
-        if (peer_state->m_local_set.size() >= MAX_RECONSET_SIZE) return AddToSetResult::Failed();
+        auto set_size = peer_state->ReconSetSize();
+        if (set_size >= MAX_RECONSET_SIZE) return AddToSetResult::Failed();
 
         // The caller currently keeps track of the per-peer transaction announcements, so it
         // should not attempt to add same tx to the set twice. However, if that happens, we will
         // simply ignore it.
-        if (peer_state->m_local_set.insert(wtxid).second) {
+        if (peer_state->m_delayed_local_set.insert(wtxid).second) {
             peer_state->m_short_id_mapping.emplace(short_id, wtxid);
             LogPrintLevel(BCLog::TXRECONCILIATION, BCLog::Level::Debug, "Added %s to the reconciliation set for peer=%d. "
                                                                         "Now the set contains %i transactions.\n",
-                          wtxid.ToString(), peer_id, peer_state->m_local_set.size());
+                          wtxid.ToString(), peer_id, set_size + 1);
```

Comment:
c790aa0e7ebfcf0b03b4d0ef5fcc0ddfa4199943
This is probably the most important log in the entire file, and I think it's better to distinguish two set sizes here (preferably other places the size is logged too).

Reply:
I changed `ReconSetSize` to return a tuple, and updated this to read:

```
 "Now the set contains %i reconcilable transactions (plus %i delayed transactions).\n",
```

Reply:
I have also addressed this when removing

Reply:
Can this be resolved?

Reply:
yes